src/ui/login/login.js

```diff
@@ -1,73 +1,110 @@
+            </div>
+            <div className="slide">
+              <h2>Explore subfolders by moving screen around</h2>
+              <div className="slide-text">
+                <p>You can see the entire structure, not just one level</p>
+              </div>
+              <img src="src/ui/login/slide-2.png" style={{width: '100%', maxWidth: '750px'}}/>
+            </div>
+            <div className="slide">
+              <h2>Frequently used folders are BIGGER</h2>
+              <div className="slide-text">
+                <p>Every time you use a folder it becomes bigger</p>
+                <p>Old files you never use get smaller, drowning into deep memory</p>
+              </div>
+              <img src="src/ui/login/slide-3.png" style={{width: '100%', maxWidth: '750px'}}/>
+            </div>
+            <div className="slide">
+              <h2>Single place for notes and files</h2>
+              <div className="slide-text">
+                <p>In NeuralNotes, every folder has a text file for the notes</p>
+                <p>It is displayed in a built-in text editor when a folder is selected</p>
+              </div>
+              <img src="src/ui/login/slide-4.png" style={{width: '100%', maxWidth: '750px'}}/>
+            </div>
+          </div>
 
-function render(props) {
-  redirectToNotesPage = props.redirectToNotesPage;
+          <h2 style={{textAlign: 'center'}}>Sign up to NeuralNotes for free</h2>
+          <a onClick={authHandler} className="btn btn-social btn-lg btn-google"
+             style={{width: '250px', display: 'block', margin: '1em auto'}}>
+            <span className="fa fa-google"/>
+            Sign up with Google
+          </a>
+        </div>
+      </div>
+    );
+  }
 
-  element = document.createElement('div');
-  element.innerHTML = loginPageHTML;
+  componentWillMount() {
+    googleApiLoader.load().then(() => this.setState({googleApiInit: true}));
+  }
 
-  onRender();
-  return element;
-}
+  /**
+   * Initiate auth flow in response to user clicking authorize button.
+   *
+   * @param {Event} event Button click event.
+   */
+  handleAuthClick = () => {
+    console.debug('login.handleAuthClick()');
+    let spinnerName = 'Loading google auth';
+    siteGlobalLoadingBar.show(spinnerName);
 
-function onRender() {
-  let authorizeButton = element.querySelector('#authorize-button');
-  authorizeButton.style.display = 'none';
-  googleApiLoader.load().then(function () {
-    authorizeButton.style.display = 'inline-block';
-    authorizeButton.addEventListener('click', handleAuthClick);
-  });
-}
+    gapiAuthorize()
+      .then(() => {
+        console.debug('login.js: auth success! calling thoughtStorage.scanDrive()');
+        siteGlobalLoadingBar.hide(spinnerName);
+      })
+      .then(googleDriveApi.loadDriveApi)
+      .then(thoughtStorage.scanDrive)
+      .then(() => {
+        console.info('login: drive scanned, redirecting to the app main page');
+        this.props.authorized();
```

Comment:
I'd call it something like `onLoginSuccess` or `onAuthSuccess` or `handleAuthSuccess`.

`authorized` is quite ambiguous: one might think it's a boolean or a function which returns a boolean.

Reply:
done